contracts/move_funds/src/lib.rs

```diff
@@ -0,0 +1,48 @@
+extern crate bincode;
+extern crate solana;
+
+use bincode::deserialize;
+use solana::system_contract::AccountInfo;
+
+#[no_mangle]
+pub extern "C" fn process(infos: &mut Vec<AccountInfo>, data: &[u8]) {
+    let tokens: i64 = deserialize(data).unwrap();
+    if infos[0].account.tokens >= tokens {
+        infos[0].account.tokens -= tokens;
+        infos[1].account.tokens += tokens;
+    } else {
+        println!(
+            "Insufficient funds, asked {}, only had {}",
+            tokens, infos[0].account.tokens
+        );
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use bincode::serialize;
+    use solana::bank::Account;
+    use solana::signature::Pubkey;
+
+    #[test]
+    fn test_move_funds() {
+        let tokens: i64 = 100;
+        let data: Vec<u8> = serialize(&tokens).unwrap();
+        let keys = vec![Pubkey::default(); 2];
+        let mut accounts = vec![Account::default(), Account::default()];
+        accounts[0].tokens = 100;
+        accounts[1].tokens = 1;
+
+        {
+            let mut infos: Vec<AccountInfo> = Vec::new();
+            for (key, account) in keys.iter().zip(&mut accounts).collect::<Vec<_>>() {
+                infos.push(AccountInfo { key, account });
+            }
```

Comment:
We use the suffix `Info` throughout the codebase to mean something like `Meta`. So the name `AccountInfo` suggests isn't information/metadata about that account, which it's not. How about `OwnedAccount`? Suggesting it's an Account with some additional information (the public key of its owner). Or maybe `AddressedAccount`?  Or perhaps punt on a name and leave it as a tuple?

Also, that code can be written without mut:

```rust
let owned_accounts: Vec<_> = keys.into_iter().zip(&accounts).map(|(key, account)| OwnedAccount {key, account}).collect();
```

Reply:
Awesome, was looking fror a one-liner for the zip :-)

For the name, maybe KeyedAccount?

Reply:
no strong opinion. I'd lean toward the tuple. :)